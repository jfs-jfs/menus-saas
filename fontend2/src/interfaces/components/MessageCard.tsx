import { Text, StyleSheet, View } from "react-native";

const styles = StyleSheet.create({
    card: {
        padding: 20,
        borderRadius: 10,
        backgroundColor: "#f1f1f1",
    },
    text: {
        fontSize: 18,
    },
});

export default function MessageCard({ text }: { text: string }) {
    return (
        <View style={styles.card}>
            <Text className="pene" style={styles.text}>{text}</Text>
        </View>
    );
}
