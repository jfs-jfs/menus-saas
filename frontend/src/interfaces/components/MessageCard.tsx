import { Text, StyleSheet, View } from "react-native";

export default function MessageCard({ text }: { text: string }) {
    return (
        <View style={styles.card}>
            <Text style={styles.text}>{text}</Text>
        </View>
    );
}

const styles = StyleSheet.create({
    card: {
        padding: 20,
        borderRadius: 10,
        backgroundColor: "#f1f1f1",
    },
    text: {
        fontSize: 18,
    },
});