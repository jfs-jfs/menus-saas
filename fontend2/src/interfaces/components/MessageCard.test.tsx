import * as React from 'react';
import {act} from 'react';
import MessageCard from '@/src/interfaces/components/MessageCard';
import { Text } from '@/src/interfaces/components/Themed';
import renderer from 'react-test-renderer';

describe('MessageCard',  () => {
    it("renders correctly with given text", () => {
      act(() => {

        const tree = renderer.create(<MessageCard text="Hello World" />).toJSON();
        expect(tree).toMatchSnapshot();
      });
      });

    it("renders the passed text", async () => {
        act(() => {
            testRenderer = renderer.create(<MessageCard text="Test Message" />);
        });
        // const testRenderer = renderer.create(<MessageCard text="Test Message" />);
        const testInstance = testRenderer.root;

        expect(testInstance.findByPropos({
          className: "pene"
        }).children).toBe("Test Message");
      })
});
