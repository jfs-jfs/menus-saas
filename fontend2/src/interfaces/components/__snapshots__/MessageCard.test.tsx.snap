// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageCard renders correctly with given text 1`] = `null`;
